cmd/object-handlers.go

```diff
@@ -256,10 +265,23 @@ func (api objectAPIHandlers) HeadObjectHandler(w http.ResponseWriter, r *http.Re
 		writeErrorResponseHeadersOnly(w, apiErr)
 		return
 	}
-	if IsSSECustomerRequest(r.Header) {
-		objInfo, err = DecryptSSECustomerObjectInfo(objInfo)
+	switch objInfo.IsEncrypted() {
```

Comment:
Perhaps this repeated function can become a function? like `checkEncrypted` ?

Reply:
Also thought about that but the HEAD must do the extra check whether the client key is correct while the GET checks this later (after range parsing) - Let me see if I can unify them...